File: signed-header-requests/signedRequest.php
<?php

function makeSignedAuthHeader($endpoint, $params, $datetime, $client_id, $secret) {
    $param_keys = array_keys($params);
    sort($param_keys);
    $kv_params = [];
    foreach($param_keys as $key) {
        array_push($kv_params, "{$key}={$params[$key]}");
    }
    $param_string = implode($kv_params, "\n");
    $string_to_sign = "${endpoint}\n${datetime}\n${param_string}\n";
    $signature = base64_encode(hash_hmac("sha1", $string_to_sign, $secret, True));
    return array("Authorization" => "Signature ${client_id}:${signature}");
}

print_r(makeSignedAuthHeader(
    '/entity.find',
    array('entity_type' => 'user', 'filter' => "lastUpdated >= '2016-01-01'"),
    'Fri, 26 Feb 2016 19:08:44 GMT',
    'apkrahlfumwse2e9nvrrotv6vchuptzw',
    'rylicq8ydkz0vmki3gqaoxbk4gyrr05t'
));

?>

File: signed-header-requests/signed-request.py
import hmac
from base64 import b64encode
from hashlib import sha1


def make_signed_auth_header(endpoint, params, datetime, client_id, secret):
    kv_params = ['{}={}'.format(k, v) for k, v in params.items()]
    kv_params.sort()
    kv_string = '\n'.join(kv_params)
    str_to_sign = '{}\n{}\n{}\n'.format(endpoint, datetime, kv_string)
    hashed_str = b64encode(hmac.new(secret, str_to_sign, sha1).digest())
    return {'Authorization': 'Signature {}:{}'.format(client_id, hashed_str)}

if __name__ == "__main__":
    print(make_signed_auth_header(
        '/entity.find',
        { 'entity_type': 'user', 'filter': "lastUpdated >= '2016-01-01'" },
        'Fri, 26 Feb 2016 19:08:44 GMT',
        'apkrahlfumwse2e9nvrrotv6vchuptzw',
        'rylicq8ydkz0vmki3gqaoxbk4gyrr05t'
    ))
